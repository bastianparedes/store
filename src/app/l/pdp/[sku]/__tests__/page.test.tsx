import { render } from '@testing-library/react';
import { redirect } from 'next/navigation';

import mockDrizzle from '../../../../../../__mocks__/drizzle';
import Page from '../page';

jest.mock('../../../../../../lib/drizzle', () => mockDrizzle);
jest.mock('next/navigation');

describe('Page', () => {
  it('should render', async () => {
    const { container } = render(await Page({ params: { sku: '1' } }));
    expect(container).toMatchSnapshot();
  });

  it('should redirect when sku is NaN', async () => {
    render(await Page({ params: { sku: 'a' } }));
    expect(redirect).toHaveBeenCalled();
  });
});
