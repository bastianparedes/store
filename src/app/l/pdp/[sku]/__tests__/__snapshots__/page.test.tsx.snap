// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="product"
    >
      <div
        class="imageContainer"
      >
        <img
          alt="name1"
          src="/api/image/1"
        />
      </div>
      <div
        class="productInfo"
      >
        <h1>
          name1
        </h1>
        <p>
          description1
        </p>
        <p>
          SKU: 
          1
        </p>
        <p>
          Cantidad disponible: 
          1
        </p>
        <p>
          Precio: $
          1000
        </p>
      </div>
    </div>
  </div>
</div>
`;
